lisa/transformer.py

```diff
@@ -174,7 +205,10 @@ def _run_transformers(
     for value in runbook_builder.variables.values():
         copied_variables[value.name] = value.copy()
 
-    factory = subclasses.Factory[Transformer](Transformer)
+    if node:
+        factory = subclasses.Factory[Transformer](DeploymentTransformer)
```

Comment:
I feel it doesn't need a different factory. If the `__init__` accepts `**kwargs`. The unused `node` parameter doesn't hurt other transformers.

Reply:
Without a different factory, it won't run the __init__ logic in DeploymentTransformer, right?

Reply:
Did you have a chance to test it? The `create_by_runbook` uses kwargs, and it passes through all named arguments to `__init__`, so it should be the same. If a transformer doesn't use node, but it's passed in, there is no error. When a transformer requires the node, but it's not passed in, we can raise an excpetion.